lib/uncruft/deprecate_method.rb

```diff
@@ -0,0 +1,25 @@
+module Uncruft
+  module DeprecateMethod
+    extend ActiveSupport::Concern
+
+    module ClassMethods
+      def deprecate_method(method, message:)
```

Comment:
I think this would still make sense with the prior name -- `deprecate_attribute` -- since it deprecates both the setter and the getter. But if you wanted, you could define it as this:

```ruby
deprecate_method method, message
deprecate_method :"#{method}=, message
```

So that's making use of a more generic `deprecate_method` method that could be defined something like this:

```ruby
prepended_method = Module.new
  define_method method do |*args, &block|
    ActiveSupport::Deprecation.warn(message)
    super(*args, &block) # passing all args straight through so that deprecation does not care about method arity
  end
  prepend prepended_method
end
```

If both `deprecate_method` and `deprecate_attribute` are public, then they can both be used by downstream consumers.